python/ray/_raylet.pyx

```diff
@@ -1398,20 +1393,32 @@ async def execute_streaming_generator_async(
             raise
         except Exception as e:
             output_or_exception = e
+            is_exception = 1
 
         loop = asyncio.get_running_loop()
         worker = ray._private.worker.global_worker
-        # Run it in a separate thread to that we can
-        # avoid blocking the event loop when serializing
-        # the output (which has nogil).
-        done = await loop.run_in_executor(
-            worker.core_worker.get_thread_pool_for_async_event_loop(),
-            report_streaming_generator_output,
-            output_or_exception,
-            context)
-        if done:
+
+        futures.append(
+            # Run it in a separate thread to that we can
+            # avoid blocking the event loop when serializing
+            # the output (which has nogil).
+            loop.run_in_executor(
+                worker.core_worker.get_thread_pool_for_async_event_loop(),
```

Comment:
IIUC, this won't work if we increase the thread pool size to > 1 (cuz I don't know if report_streaming_generator_output is thread-safe now). Can you add assert somewhere to make sure it is not updated?

Reply:
That's correct. Those APIs aren't thread-safe and i have discovered it in my experiments already, hence the cleanup that i started doing in https://github.com/ray-project/ray/pull/42443, to alleviate some of that (it's stacked on top of this one)